app/builders/account_builder.rb

```diff
@@ -42,14 +44,9 @@ def create_account
   end
 
   def create_and_link_user
-    password = Time.now.to_i
-    @user = User.new(email: @email,
-                     password: password,
-                     password_confirmation: password,
-                     name: email_to_name(@email))
-    @user.confirm if @confirmed
-    if @user.save!
+    if @user.present? || create_user
       link_user_to_account(@user, @account)
+      @user.linked_account_id = @account.id
```

Comment:
I think this is not required, you can just return the account which is created instead of a linked account id. @sojan-official requesting a review here.